src/components/Dropdown/Dropdown.types.ts

```diff
@@ -11,6 +12,14 @@ export interface DropdownProps {
      * @param visible {boolean}
      */
     onVisibleChange?: (visible: boolean) => void;
+    /**
+     * Callback called when the visibility of the dropdown changes
+     * @param visible {boolean}
+     */
+    onToggle?: (
+        currentState: ToggleState,
+        changes: Partial<ToggleState>
+    ) => ToggleState;
```

Comment:
@ychhabra-eightfold 
at present dropdown is an uncontrolled component. 
In case of multi select flow, dropdown should stay visible even after a selection.
But this is not possible at present with the dropdown component because we toggle the visible state on click
to handle that we are allowing the parent component to decide whether to close or open.

`onVisibleChange`  just notifies that there is a change in the visibility 
but the above talks about whether to have a change.

Let me know if I makes sense, we can sync over this